client-spark/spark3/src/main/java/org/apache/spark/shuffle/writer/RssShuffleWriter.java

```diff
@@ -289,25 +291,27 @@ public void write(Iterator<Product2<K, V>> records) {
   private void writeImpl(Iterator<Product2<K, V>> records) {
     List<ShuffleBlockInfo> shuffleBlockInfos;
     boolean isCombine = shuffleDependency.mapSideCombine();
-    Function1<V, C> createCombiner = null;
+
+    Iterator<? extends Product2<K, ?>> iterator = records;
     if (isCombine) {
-      createCombiner = shuffleDependency.aggregator().get().createCombiner();
+      if (RssSparkConfig.toRssConf(sparkConf).get(RSS_CLIENT_MAP_SIDE_COMBINE_ENABLED)) {
+        iterator = shuffleDependency.aggregator().get().combineValuesByKey(records, taskContext);
```

Comment:
Do we need to check the existence of `shuffleDependency.aggregator()` ?

Reply:
BTW, will this map combine cause the disk/memory burden , especially on the tight disk space on k8s?

Reply:
> Do we need to check the existence of shuffleDependency.aggregator() ?

It seems not, there is a relevant check in ShuffleDependency.

https://github.com/apache/spark/blob/2ac2710b46be70064cd7286a9c86deb1ddc979cb/core/src/main/scala/org/apache/spark/Dependency.scala#L89-L91

Reply:
Good to know this.

Reply:
> BTW, will this map combine cause the disk/memory burden , especially on the tight disk space on k8s?

I think it is possible when there are spills. So I added a configuration to control whether it is enabled.

Reply:
Yeh. I got your point, just an extra attention. 

For having this point implicitly, let's make this shown in this config description.